src/testdir/test_visual.vim

```diff
@@ -1630,4 +1630,118 @@ func Test_visual_substitute_visual()
   bwipe!
 endfunc
 
+func Test_visual_getregion()
+  new
+
+  call setline(1, ['one', 'two', 'three'])
+
+  " Visual mode
+  call cursor(1, 1)
+  call feedkeys("\<ESC>vl", 'tx')
+  call assert_equal(['on'], 'v'->getregion('.', 'v'))
+  call assert_equal(['on'], '.'->getregion('v', 'v'))
+  call assert_equal(['o'], 'v'->getregion('v', 'v'))
+  call assert_equal(['n'], '.'->getregion('.', 'v'))
+
+  " Line visual mode
+  call cursor(1, 1)
+  call feedkeys("\<ESC>Vll", 'tx')
+  call assert_equal(['one'], getregion('v', '.', 'V'))
+  call assert_equal(['one'], getregion('.', 'v', 'V'))
+  call assert_equal(['one'], getregion('v', 'v', 'V'))
+  call assert_equal(['one'], getregion('.', '.', 'V'))
+
+  " Block visual mode
+  call cursor(1, 1)
+  call feedkeys("\<ESC>\<C-v>ll", 'tx')
+  call assert_equal(['one'], getregion('v', '.', "\<C-v>"))
+  call assert_equal(['one'], getregion('.', 'v', "\<C-v>"))
+  call assert_equal(['o'], getregion('v', 'v', "\<C-v>"))
+  call assert_equal(['e'], getregion('.', '.', "\<C-v>"))
+
+  " Multiline with line visual mode
+  call cursor(1, 1)
+  call feedkeys("\<ESC>Vjj", 'tx')
+  call assert_equal(['one', 'two', 'three'], getregion('v', '.', 'V'))
+
+  " Multiline with block visual mode
+  call cursor(1, 1)
+  call feedkeys("\<ESC>\<C-v>jj", 'tx')
+  call assert_equal(['o', 't', 't'], getregion('v', '.', "\<C-v>"))
+
+  call cursor(1, 1)
+  call feedkeys("\<ESC>\<C-v>jj$", 'tx')
+  call assert_equal(['one', 'two', 'three'], getregion('v', '.', "\<C-v>"))
```

Comment:
Can you add a test for using getregion() with "V" but the text is selected using block-wise visual mode.  And additional tests for other combinations?

Reply:
I have added them.